doc/introduction.rst

```diff
@@ -0,0 +1,31 @@
+============
+Introduction
+============
+
+Distance metrics are widely used in the machine learning literature.
+Traditionally, practitioners would choose a standard distance metric
+(Euclidean, City-Block, Cosine, etc.) using a priori knowledge of
+the domain.
+Distance metric learning (or simply, metric learning) is the sub-field of
+machine learning dedicated to automatically constructing optimal distance
+metrics.
```

Comment:
> * "optimal" is not great (wrt what is it optimal?). I would instead say "to automatically construct task-specific distance metrics from (weakly) supervised data".
> 
I agree, done

> * the new quick start example below makes me realize that we should clearly mention the relation between distance metric learning and embedding/representation learning. We could add a sentence like:
>   "The learned distance metric often corresponds to a Euclidean distance in a new embedding space, hence distance metric learning can be seen as a form of representation learning."

I agree, for now I just added this sentence, but indeed we should emphasize this more, maybe in the future with some section in the docs, and also maybe this will get clearer with examples of metric learners used as transformers (examples of dimensionality reduction for instance)